src/controllers/controller.cpp

```diff
@@ -37,6 +37,18 @@ ControllerJSProxy* Controller::jsProxy() {
     return new ControllerJSProxy(this);
 }
 
+QString Controller::physicalTransport2String(PhysicalTransportProtocol protocol) {
+    static const QMap<PhysicalTransportProtocol, QString> protocolMap = {
+            {PhysicalTransportProtocol::USB, QStringLiteral("USB")},
+            {PhysicalTransportProtocol::BlueTooth, QStringLiteral("Bluetooth")},
+            {PhysicalTransportProtocol::I2C, QStringLiteral("I2C")},
+            {PhysicalTransportProtocol::SPI, QStringLiteral("SPI")},
+            {PhysicalTransportProtocol::FireWire, QStringLiteral("Firewire - IEEE 1394")},
+            {PhysicalTransportProtocol::UNKNOWN, tr("Unknown")}};
+
+    return protocolMap.value(protocol, tr("Unknown"));
```

Comment:
why the map over a `switch-case`?

Reply:
It doesn't matter here, but in general, a map scales better in lookup performance. So it's just use of the common pattern.

Reply:
> It doesn't matter here, but in general, a map scales better in lookup performance. So it's just use of the common pattern.

I actually highly doubt that when they keys are effectively `int`s.